Import nickel-nix parts from nickel-nix input

YorikSar opened the pull request:
Move everything from `nixel` repo in templates to the `nickel-nix` repo itself and import from it as input in the templates.

This will require user to run `nix run .#generate-lockfile` after flake initialisation and on each change of the flake lock file.

Fixes #24

thufschmitt commented on templates/devshells/c/flake.nix:
```diff
@@ -35,5 +34,13 @@
+      apps = {
+        regenerate-lockfile = {
+          type = "app";
+          program = pkgs.lib.getExe (buildLockFile {
```
I'm not sure why, but this `getExe` seems to be the thing causing the CI failure.

thufschmitt replied:
Gne, nevermind that one, it's Friday

thufschmitt replied:
The actuall issue is

```suggestion
          program = pkgs.lib.getExe (nickel-nix.lib.${system}.buildLockFile {
```

YorikSar replied:
It's already imported in let block above, so this shouldn't be the issue...

YorikSar commented on templates/devshells/c/flake.nix:
```diff
@@ -35,5 +34,13 @@
       devShells = {
         default = nickelDerivation;
       };
+      apps = {
+        regenerate-lockfile = {
+          type = "app";
+          program = pkgs.lib.getExe (buildLockFile {
+            nickel-nix = nickel-nix.lib.${system}.lockFileContents;
+          });
+        };
+      };
```
Good idea. Will add this.

YorikSar commented:
@thufschmitt Since your review I have:
* added a higher level wrapper that provides the "flake app" directly, so that boilerplate is gone
* made `nickel-nix-internals` input implicit, but using user's `nixpkgs`, so user doesn't need to know about naked stdenv
* fixed tests: they were using wrong version of nickel-nix input, so moved override to separate call of `nix flake lock`
* added topiary's cache to all flakes to not rebuild it

Please take another look.

thufschmitt reviewed (approved):
Thanks, and sorry for the delay in the review
